eldritch/stories.py

```diff
@@ -0,0 +1,202 @@
+
+  def get_trigger(self, event, owner, state):
+    if (
+        isinstance(event, events.TakeGateTrophy)
+        and (event.character == owner)
+        and (event.gate.name == "Dreamlands")
+    ):
+      return self.advance_story(owner, True)
+    if (
+        isinstance(event, events.AddDoom)
+        and state.ancient_one.doom >= 5
+    ):
+      return self.advance_story(owner, False)
+    return super().get_trigger(event, owner, state)
+
+
+class GangsterStoryPass(StoryResult):
+  def __init__(self):
+    super().__init__("This One's for Louis", {"max_sanity": 1})
+
+  def get_in_play_event(self, owner: characters.Character):
+    print("Gangster Story Pass now in play")
+    return events.Gain(owner, {"sanity": 1}, self)
+
+  def get_trigger(self, event, owner, state):
+    if isinstance(event, events.TakeTrophy) and event.character == owner:
+      return events.Gain(owner, {"sanity": 1}, self)
+    return super().get_trigger(event, owner, state)
+
+
+class GangsterStoryFail(StoryResult):
+  def __init__(self):
+    super().__init__("Crime Doesn't Pay")
+
+  def get_in_play_event(self, owner: characters.Character):
+    return events.Loss(owner, {"dollars": owner.dollars})
+
+  def get_interrupt(self, event, owner, state):
+    if (
+        isinstance(event, events.GainOrLoss)
+        and event.character == owner
+        and "dollars" in event.gains
+    ):
+      return events.GainOrLossPrevention(self, event, "dollars", float("inf"), "gains")
+    return None
+
+
+class GangsterStory(Story):
+  def __init__(self):
+    super().__init__("For a Friend", "This One's for Louis", "Crime Doesn't Pay")
+
+  def get_trigger(self, event, owner: characters.Character, state):
+    n_trophies = len([t for t in owner.trophies if isinstance(t, monsters.Monster)])
+    if isinstance(event, events.ForceTakeTrophy) and n_trophies >= 5:
+      return self.advance_story(owner, True)
+    if (
+        isinstance(event, events.GainOrLoss)
+        and (event.character == owner)
+        and (event.gains.get("dollars", 0) >= 5)
+    ):
+      return self.advance_story(owner, False)
+    return super().get_trigger(event, owner, state)
+
+
+class NunStoryPass(StoryResult):
+  def __init__(self):
+    super().__init__("Fear No Evil")
+
+  def get_interrupt(self, event, owner, state):
+    if isinstance(event, events.Curse) and event.character == owner:
+      return events.CancelEvent(event)
+    return super().get_interrupt(event, owner, state)
+
+  def get_usable_trigger(self, event, owner, state):
+    if isinstance(event, events.DiceRoll) and state.turn_phase == "upkeep":
+      # Note: I see nothing that says it has to be one of her dice
+      # TODO: Allow user to select a die to reroll
+      to_reroll = [0]
+      # to_reroll = events.ChooseDie(event)
+      return events.Sequence([
+          events.ExhaustAsset(owner, self),
+          events.RerollSpecific(state.event_stack[-2].character, state.event_stack[-2], to_reroll),
+      ], owner)
```

Comment:
Oof, RerollSpecific currently only works for checks; you may have to update it to work for either checks or DiceRolls.

Reply:
Didn't feel like tracing all those threads, so I just made a new Event. There's not a specific test for the new event type, but the Nun story test does test it.